main.py

```diff
@@ -337,6 +342,50 @@ def isPaused(self):
 	def isRunning(self):
 		return self._p is not None # FIXME: Check subproc state
 
+class AppVeyor(object):
+	def __init__(self, repo, branch, user):
+		self.gjson = json.loads(requests.get("https://ci.appveyor.com/api/projects/" + user + "/" + repo + "/branch/" + branch).text)
+
+	def GetJobs(self):
+		return self.gjson["build"]["jobs"]
+
+	def GetRelease(self, id):
+		idjson = json.loads(requests.get("https://ci.appveyor.com/api/buildjobs/" + str(id) + "/artifacts").text)
+		return idjson
+
+class Updater(object):
+    def __init__(self, settings):
+        self.gsettings = settings
+        appveyor = AppVeyor("xqemu", "master", "xqemu-bot")
+        self.jobs = appveyor.GetJobs()
+        self.release = appveyor.GetRelease(self.jobs[0]["jobId"])
```

Comment:
This is almost never used (only for getting the size?)

Even worse: when it is used, it's used decoupled from `self.jobs[0]`.
So if we change it to `self.jobs[1]` here, we'll actually display the wrong information / download another build than we pretend we do.

Try to keep these things close to each other and almost impossible to get out of sync. If you only need the release in 1 function, probably it's best to download the info in that function (if it uses internet access anyway).

Alternatively, hide the option which might lead to this misalignment: Instead of storing `self.jobs`, you could only store `self.job` - however, I think that's also a bad idea.
You could also do something like:

```
self.releases = {}
job = self.jobs[0] # In the future we might download release info for all job-ids by iterating over them
self.releases[job["jobId"]] = appveyor.GetRelease(job["jobId"])
```

And when using it, make sure that the release info exists in cache already.

```
if job["jobId"] is not in self.release:
    # Throw error about not being able to find valid release information, "Please try again"
    return
```